### ISSUE TITLE
vfs.DirExists behaves differently for ProjectedFilesystem

### ISSUE THREAD

**Issue by Diaphteiros** [mandelsoft/vfs#3]
There is a bug in the implementation for projected filesystems:

Example code:
```golang
package main

import (
	"fmt"
	"path/filepath"

	"github.com/mandelsoft/vfs/pkg/memoryfs"
	"github.com/mandelsoft/vfs/pkg/osfs"
	"github.com/mandelsoft/vfs/pkg/projectionfs"
	"github.com/mandelsoft/vfs/pkg/vfs"
)

func main() {
	path := "/tmp/foo" // this is an existing but empty directory on the host's filesystem

	baseFs := osfs.New()
	exists, err := vfs.DirExists(baseFs, filepath.Join(path, "bar", "baz"))
	fmt.Println(exists, err)

	memFs := memoryfs.New()
	exists, err = vfs.DirExists(memFs, filepath.Join(path, "bar", "baz"))
	fmt.Println(exists, err)

	proFs, _ := projectionfs.New(baseFs, "/")
	exists, err = vfs.DirExists(proFs, filepath.Join(path, "bar", "baz"))
	fmt.Println(exists, err)
}
```

Result:
```
false <nil>
false <nil>
false stat /tmp/foo/bar/baz: lstat /private/tmp/foo/bar: no such file or directory
```

`vfs.DirExists` called for a non-existing directory with a non-existing direct parent returns an error if the underlying filesystem is a projected filesystem. A check whether a directory exists should not return an error if it (or any of its parents) does indeed not exist. The same function works as expected for os and memory filesystems.

**Comment by Diaphteiros** [comment]
I found another small deviation, although this one is not as relevant (but it might be connected to the other issue):
```golang
	path := "/tmp/foo/baz/bam"
	// neither 'baz' nor 'bam' exist

	_, err := os.Stat(path)
	fmt.Println(err, os.IsNotExist(err), errors.Is(err, os.ErrNotExist))

	baseFs := osfs.New()
	_, err = baseFs.Stat(path)
	fmt.Println(err, os.IsNotExist(err), errors.Is(err, os.ErrNotExist))

	proFs, _ := projectionfs.New(baseFs, "/")
	_, err = proFs.Stat(path)
	fmt.Println(err, os.IsNotExist(err), errors.Is(err, os.ErrNotExist))
```

Result:
```
stat /tmp/foo/baz/bam: no such file or directory true true
stat /tmp/foo/baz/bam: no such file or directory true true
stat /tmp/foo/baz/bam: lstat /private/tmp/foo/baz: no such file or directory false true
```

This small example shows that for a missing parent directory, `Stat` returns an error which is recognized as `ErrNotExist` by the `os.IsNotExist` function when called for the golang `os` package and for an os filesystem, but when called on a projection filesystem, the error is not recognized by `os.IsNotExist` anymore. It is correctly recognized by `errors.Is(err, os.ErrNotExist)` in all cases though, which is the recommended way of checking this anyway.

**Comment by mandelsoft** [comment]
Fix with 3927141.
